ngraph/frontend/paddlepaddle/src/op/elementwise_ops.cpp

```diff
@@ -0,0 +1,82 @@
+//*****************************************************************************
+// Copyright 2017-2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <map>
+
+#include <ngraph/opsets/opset6.hpp>
+#include "elementwise_ops.hpp"
+
+namespace ngraph {
+namespace frontend {
+namespace pdpd {
+namespace op {
+
+template <typename T>
+OutputVector elementwise_ops (const NodeContext& node, const T& def = T()) { 
+    auto x = node.get_ng_input("X");
+    auto y = node.get_ng_input("Y");    
+
+    auto axis = node.get_attribute<int>("axis");
+
+    auto x_shape = x.get_shape();
+    auto y_shape = y.get_shape();
+
+    if ((axis == -1) || (axis == x_shape.size() - 1) || (x_shape.size() == y_shape.size())) {
+        return {std::make_shared<T>(x, y)};
+    }
+    else {
+        auto broadcast_shape = Shape(x_shape.size(), 1);
+        int32_t idx = 0;
+        for(auto it = y_shape.begin(); it != y_shape.end(); ++it, idx++)
+            broadcast_shape[axis+idx] = *it;
```

Comment:
Please leave a link in the code comments to the broadcast specification that you are implementing here. Can we just use Broadcast op https://docs.openvinotoolkit.org/latest/openvino_docs_ops_movement_Broadcast_3.html?

Reply:
It looks both Reshape and Broadcast can resolve the shape-mismatching issue of ElementWise_ops. 

This code snippet refers to Paddle2ONNX, where Reshape is used -https://github.com/PaddlePaddle/Paddle2ONNX/blob/71c5b1688838b48fa2c227bce511bac08fb919da/paddle2onnx/op_mapper/math.py#L122 

The Reshape solution is based on the fact that - y_shape is a strict part of x_shape, while the Broadcast solution has no such requirement. 

Problem is, which is better? 

Perhaps Reshape is, since it would be just a "reshape" literally, without actual data movement involved in plugin implementation. Correct me if my understanding is wrong.